src/fetch/transfer_file.py

```diff
@@ -37,10 +39,9 @@ def manual_propose(dune: DuneFetcher) -> None:
         f"{dune.period.unusual_slippage_url()}"
     )
     transfers = Transfer.consolidate(dune.get_transfers())
-    write_to_csv(
-        data_list=[CSVTransfer.from_transfer(t) for t in transfers],
-        outfile=File(name=f"transfers-{dune.period}.csv"),
-    )
+    csv_transfers = [asdict(CSVTransfer.from_transfer(t)) for t in transfers]
+    FileIO(FILE_OUT_DIR).write_csv(csv_transfers, f"transfers-{dune.period}.csv")
```

Comment:
Discussion: Why do we have to pass both the file_out_dir and filename?

Reply:
Generally that is how file writing works and also how this is implemented:

One specifies the "directory" and the "filename". 

If one tries to combine the two it can result in less robust functionality (issues with the use of relative paths, etc...)

The `FileIO` class is being used also by our Dune Sync, where the user creates one "instance" specifying the path that all files will be written to. then calls write whenever they want with many different filenames and because we use the same instance the path does not need to be specified over and over again.

Its kinda like saying, at the start "all files will be written to path" then the path does not need to be set again.

Anyway, happy to discuss more, but I think it is common practice when writing IO stuff to separate the path from the file name.

Reply:
> Generally, I believe, that is how file writing works and also how the class `FileIO` is implemented:
> 
> One specifies the "directory" and the "filename".
> 
> If one tries to combine the two it can result in less robust functionality (issues with the use of relative paths, etc...)
> 
> The `FileIO` class is being used also by our Dune Sync, where the user creates one "instance" specifying the path that all files will be written to. then calls write whenever they want with many different filenames and because we use the same instance the path does not need to be specified over and over again.
> 
> Its kinda like saying, at the start "all files will be written to path" then the path does not need to be set again.
> 
> Anyway, happy to discuss more, but I think it is common practice when writing IO stuff to separate the path from the file name.

Thanks for the explanation. Appreciate the effort you've put into it.

It makes sense what you've said. Just coming from pandas there you usually just call `.to_csv()` and can specify the filename and path as part of that there.